app/routes.py

```diff
@@ -1,2 +1,338 @@
+@rentals_bp.route("/check-in", methods=["POST"], strict_slashes=False)
+# ❗️ I chose to put this in a separate endpoint, wondering if that's correct or if I should have just integrated this code into the check-out endpoint function 
+def check_in_video():
+
+    request_body = request.get_json()
+
+    # ❗️ This is such a clunky way to go about this, I just ran out of time. I was trying to figure out a way to mark a rental record as checked in without physically deleting it from the rentals table. This code checks if a rental record is missing a due date, in which case it's presumed that the respective video was already checked in. Super obviously problematic logic, but the code passed the tests. I probably could have just made another attribute for the rental record or something, and then put all of this in a helper method - just ran out of time.
+    # Checks if there is even a rental record for this request, so a customer can't check a video back in twice
+    video_id = request_body["video_id"]
+    customer_id = request_body["customer_id"]
+    rental_records = Rental.query.filter_by(fk_video_id=video_id, fk_customer_id=customer_id)
+    for each_rental in rental_records:
+        record_due_date = each_rental.due_date
+        rental_record_id = each_rental.rental_id
+
+    # if the record due date is None:
+    if not record_due_date:
+        return make_response({"details": f"Video with video id {video_id} was already checked in"}, 400)
+    
+    updated_cust = Customer.query.get_or_404(request_body["customer_id"])
+    updated_cust.videos_checked_out_count -= 1
+
+    updated_video = Video.query.get_or_404(request_body["video_id"])
+    updated_video.available_inventory += 1
+
+    # Once a video gets checked in, this sets the rental record for that video/customer to None so the video cannot be checked back in
+    rental = Rental.query.get_or_404(rental_record_id)
+    rental.due_date = None 
+
+    db.session.commit()
+
+    return make_response({ "customer_id": updated_cust.cust_id,
+                        "video_id": updated_video.video_id,
+                        "videos_checked_out_count": updated_cust.videos_checked_out_count,
+                        "available_inventory": updated_video.available_inventory
+                    }, 
+                        200)
+
+    
+@customers_bp.route("/<customer_id>/rentals", methods=["GET"], strict_slashes=False)
+def get_customer_rentals(customer_id):
+
+    current_rentals = []
+    current_rentals_response = []
+
+    # ❗️ concerned again about the fact that I'm not really updating the rental table at any point to indicate whether its records are current or not....
+    rental_records = Rental.query.filter_by(fk_customer_id=customer_id)
+
+    # ❗️ I know this can't be the best way to do this (might even be...the worst way), but trying to use `saved_customer.current_rentals` to get some kind of list of current rentals wasn't working in the way I expected, and I just ran out of time 
```

Comment:
Huh, bummer you couldn't get that attribute to work! but great that you got this working regardless!

Reply:
Again, I really don't think this is bad, even if it is doing a little unnecessary work!